server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -237,9 +267,13 @@ public void cancel(CancellableTask task, String reason, Runnable listener) {
     public Task unregister(Task task) {
         logger.trace("unregister task for id: {}", task.getId());
 
-        // Decrement the task's self-thread as part of unregistration.
+        // Decrement the task's self-thread as part of un-registration.
         task.decrementResourceTrackingThreads();
 
+        if (taskResourceConsumersEnabled && task.getResourceStats().isEmpty() == false) {
+            taskResourceConsumer.forEach(consumer -> consumer.accept(task));
+        }
```

Comment:
Lets add exception handling so that we don't fail forth coming consumers if the current consumer failed to work.

Reply:
Added try catch block to handle any exception thrown from consumers.